packages/lisk-p2p/test/unit/response_handler.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { expect } from 'chai';
+import {
+	checkForValidPeerAddress,
+	instantiatePeerFromResponse,
+} from '../../src/response_handler';
+
+describe('response handlers', () => {
+	describe('#instantiatePeerFromResponse', () => {
+		describe('for valid peer response object', () => {
+			const peer: unknown = {
+				ip: '12.23.54.3',
+				wsPort: '5393',
+				os: 'darwin',
+				height: '23232',
+				version: '1.1.2',
+			};
+
+			const peerWithInvalidHeightValue: unknown = {
+				ip: '12.23.54.3',
+				wsPort: '5393',
+				os: '778',
+				height: '2323wqdqd2',
+				version: '3.4.5-alpha.9',
+			};
+
+			const peerWithInvalidOsValue: unknown = {
+				ip: '12.23.54.3',
+				wsPort: '5393',
+				os: '778',
+				height: '23232',
+				version: '3.4.5-alpha.9',
+			};
+
+			it('should return PeerConfig object', () => {
+				return expect(instantiatePeerFromResponse(peer))
+					.to.be.an('object')
+					.include({
+						ipAddress: '12.23.54.3',
+						wsPort: 5393,
+						os: 'darwin',
+						height: 23232,
+						version: '1.1.2',
+					});
+			});
+
+			it('should return PeerConfig object with height value set to 0', () => {
+				return expect(instantiatePeerFromResponse(peerWithInvalidHeightValue))
+					.to.be.an('object')
+					.include({
+						ipAddress: '12.23.54.3',
+						wsPort: 5393,
+						os: '',
+						height: 0,
+						version: '3.4.5-alpha.9',
+					});
+			});
+
+			it('should return peer object and instance of Peer sets blank for invalid value of os', () => {
+				return expect(instantiatePeerFromResponse(peerWithInvalidOsValue))
+					.to.be.an('object')
+					.include({
+						ipAddress: '12.23.54.3',
+						wsPort: 5393,
+						os: '',
+						height: 23232,
+						version: '3.4.5-alpha.9',
+					});
+			});
+		});
+
+		describe('for invalid peer response object', () => {
+			it('should return false for invalid peer values', () => {
```

Comment:
You should update the text since it's now throwing instead of returning.

Reply:
addressed